CMakeLists.txt

```diff
@@ -61,60 +42,81 @@ set_target_properties(
         Fortran_MODULE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/mod"
 )
 
+#flcl-cxx library
 add_library(flcl-cxx
     OBJECT
         src/flcl-cxx.cpp
         src/flcl-util-cxx.cpp
 )
+set(flcl-cxx-public-headers
+    ${PROJECT_SOURCE_DIR}/src/flcl-cxx.hpp
+    ${PROJECT_SOURCE_DIR}/src/flcl-util-cxx.h
+)
+set_target_properties(flcl-cxx PROPERTIES PUBLIC_HEADER "${flcl-cxx-public-headers}")
 target_link_libraries(flcl-cxx
     PRIVATE
         Kokkos::kokkos
 )
 
-add_library(FLCL
+add_library(flcl
     STATIC
         $<TARGET_OBJECTS:flcl-fortran>
         $<TARGET_OBJECTS:flcl-cxx>
 )
-target_include_directories(FLCL
+set_target_properties(flcl PROPERTIES PUBLIC_HEADER "${flcl-cxx-public-headers}")
+target_include_directories(flcl
+    PUBLIC
+        $<INSTALL_INTERFACE:"${CMAKE_INSTALL_INCLUDEDIR}">
     INTERFACE
         $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/mod>
-        $<INSTALL_INTERFACE:mod>
+        #$<INSTALL_INTERFACE:mod>
 )
-target_link_libraries(FLCL
+target_link_libraries(flcl
     INTERFACE
     Kokkos::kokkos
 )
-add_library(FLCL::flcl ALIAS FLCL)
+add_library(flcl::flcl ALIAS flcl)
 
+#installation section
 include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
+configure_package_config_file(
+    "cmake/flclConfig.cmake.in"
+    "${PROJECT_BINARY_DIR}/flclConfig.cmake"
+    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake"
+)
 
 write_basic_package_version_file(
     "${PROJECT_BINARY_DIR}/flclConfigVersion.cmake"
+    VERSION "${FLCL_VERSION}"
```

Comment:
I meant you can use `${PROJECT_VERSION}` here.  I am not asking you to change though

Reply:
I'm all for making things more succinct and standard.